lib/rsolr/client.rb

```diff
@@ -319,7 +320,7 @@ def connection
         conn.request :retry, max: options[:retry_after_limit], interval: 0.05,
                              interval_randomness: 0.5, backoff_factor: 2,
                              exceptions: ['Faraday::Error', 'Timeout::Error'] if options[:retry_503]
-        conn.adapter options[:adapter] || Faraday.default_adapter
+        conn.adapter options[:adapter] || Faraday.default_adapter || :net_http
```

Comment:
Does Faraday.default_adapter return nil in 2.x? Why do they even have that method?

Reply:
According to the upgrade guide, downstream consumers can set default_adapter.